python/GafferUI/CompoundEditor.py

```diff
@@ -585,6 +656,33 @@ def __pinningButtonContextMenu( self, button ) :
 
 		return True
 
+	def __addStandardItems( self, editor, m ) :
+
+		# @see CompoundEditor.nodeSetMenuSignal for details on the structure
+		# of this menu.
+
+		drivingEditor, _ = editor.getNodeSetDriver()
+		isPinned  = self.__editorIsPinned( editor )
+
+		isFollowingNodeSelection = drivingEditor is None and not isPinned
+		m.append( "/Following Selected Nodes", {
```

Comment:
How about:

`m.append( "/Follow Node Selection", {`

I would stick to the same phrasing you used elsewhere, to avoid ambiguity where the user might think they can focus on more than one node at a time for editors that can't yet.

I'm not completely sold on "following" as being the correct verb here. Was "focus" avoided for a reason?

Reply:
Yeah, we're not entirely happy with that whole thing.

>   * I'm not sure about distinguishing solely by colour on such a small element, when colour blindness and visual acuity of aging artists is something to consider.

Any ideas for alternatives?

>   * I'm also not sure 4 colours will be enough for some very messy graphs, especially now that we've opened the door to multi-monitor layouts. It wasn't clear to me at first that when I was linking to my 5th _Viewer_ that this was a _new_ red link, and not the first one again.

We can easily add more :) how many do you think is a realistic upper bound?

Reply:
> I would stick to the same phrasing you used elsewhere, to avoid ambiguity where the user might think they can focus on more than one node at a time for editors that can't yet.

Skipping the follow/focus change for a mo, as this applies to either, I think the thinking was that the top half is state, so it describes 'how it is'. Following... Pinned... If we were to change it to 'Follow' we'd also have to change it to 'Pin'. Then when you have another item below, that is purely a one-shot action, 'Pin ...' is the different between the two the apparent? 

Youd end up with:

```
Follow Selected Nodes
Pin
Link to Editor >
---------
Pin Current Node Selection
Pin Bookmark
--------
Bookmark <current node>
```

Reply:
> Any ideas for alternatives?

The only thing that came to mind was to use a number next to the pin icon, which follows the pattern we established with bookmarked nodes. I suppose there's a potential to confuse editor number with bookmark number, but at least it will be consistent? This also solves the need to define a finite number of colours, and accommodate various people's eyesight.

> Skipping the follow/focus change for a mo...

Oops, I actually wanted to avoid splitting hairs on the infinitive/imperative distinction, but I wrote `Follow` without thinking. Freudian typing, I suppose.

I think using imperative everywhere is fine. I know it's inconsistent with other "state" menu items we've established, but those are already inconsistent, with actions like _Show Input/Output Connections_ in the node right-click menu.

But I don't want to derail you, so let's talk about UI text conventions elsewhere. For this PR, I'm more interested in "focusing" vs. "following".

Reply:
> The only thing that came to mind was to use a number next to the pin icon, which follows the pattern we established with bookmarked nodes.

The bookmark connotation you highlighted is problematic here I think. We plan to extend the pinning options with "Following Bookmark 1" etc, so any use of numbers should be reserved to that. My only other suggestion would be shape, but I think colour is probably a better compromise.

> I'm not completely sold on "following" as being the correct verb here. Was "focus" avoided for a reason?

Following implies continuous activity - i.e. the editor will adjust to follow future changes. Focussing implies a temporary transition state on the way to being focussed on one thing. We also have a need for the reverse term in tooltips : "Followed by ...", and there doesn't seem to be a good phrasing of that for "focus".

> I actually wanted to avoid splitting hairs on the infinitive/imperative distinction

I'm actually quite interested in this one, but as you say, let's discuss at leisure and looking at the UI as a whole, rather than focus on it for just this PR.